src/Spice86.Core/Emulator/InterruptHandlers/Dos/Xms/ExtendedMemoryManager.cs

```diff
@@ -0,0 +1,1887 @@
+    /// <item>Initializes the XMS memory area as a single free block</item>
+    /// </list>
+    /// </remarks>
+    public ExtendedMemoryManager(IMemory memory, State state, A20Gate a20Gate,
+        MemoryAsmWriter memoryAsmWriter, DosTables dosTables,
+        ILoggerService loggerService) {
+        uint headerAddress = MemoryUtils.ToPhysicalAddress(DosDeviceSegment, 0);
+        Header = new DosDeviceHeader(memory,
+            headerAddress) {
+            Name = XmsIdentifier,
+            StrategyEntryPoint = 0,
+            InterruptEntryPoint = 0
+        };
+        _state = state;
+        _a20Gate = a20Gate;
+        _memory = memory;
+        _loggerService = loggerService;
+        // Place hookable callback in writable memory area
+        var hookableCodeAddress = new SegmentedAddress((ushort)(dosTables
+            .GetDosPrivateTableWritableAddress(0x1) - 1), 0x10);
+        CallbackAddress = hookableCodeAddress;
+        SegmentedAddress savedAddress = memoryAsmWriter.CurrentAddress;
+        memoryAsmWriter.CurrentAddress = hookableCodeAddress;
+        memoryAsmWriter.WriteJumpNear(0x3);
+        memoryAsmWriter.WriteNop();
+        memoryAsmWriter.WriteNop();
+        memoryAsmWriter.WriteNop();
+        memoryAsmWriter.RegisterAndWriteCallback(0x43, RunMultiplex);
+        memoryAsmWriter.WriteFarRet();
+        memoryAsmWriter.CurrentAddress = savedAddress;
+        Name = XmsIdentifier;
+
+        // Initialize XMS memory as a single free block
+        _xmsBlocksLinkedList.AddLast(new XmsBlock(0, 0, XmsMemorySize * 1024, true));
+    }
+
+    /// <summary>
+    /// Gets the largest free block of memory in bytes.
+    /// </summary>
+    /// <remarks>
+    /// This property is used by Functions 08h (Query Free Extended Memory) and 88h (Query Any Free Extended Memory)
+    /// to report the size of the largest available memory block. It's calculated by finding the largest
+    /// free block in the XMS memory pool.
+    /// </remarks>
+    public uint LargestFreeBlockLength {
+        get {
+            IEnumerable<XmsBlock> elements = GetFreeBlocks();
+            if(!elements.Any()) {
+                return 0;
+            }
+            return elements.Max(static x => x.Length);
+        }
+    }
+
+    /// <summary>
+    /// Gets the total amount of free memory in bytes.
+    /// </summary>
+    /// <remarks>
+    /// This property is used by Functions 08h (Query Free Extended Memory) and 88h (Query Any Free Extended Memory)
+    /// to report the total free memory available. It's calculated by summing the sizes of all free blocks
+    /// in the XMS memory pool.
+    /// </remarks>
+    public long TotalFreeMemory {
+        get {
+            IEnumerable<XmsBlock> elements = GetFreeBlocks();
+            if (!elements.Any()) {
+                return 0;
+            }
+            return elements.Sum(static b => b.Length);
+        }
+    }
```

Comment:
```suggestion
        public long TotalFreeMemory => _xmsBlocksLinkedList
        .Where(static x => x.IsFree)
        .Sum(static b => b.Length);
```
Slightly more efficient without sorting.